### ISSUE TITLE
React Native Authenticator - SignIn field "isRequired" ignored

### ISSUE THREAD

**Issue by andreav** [aws-amplify/amplify-ui#3190]
### Before creating a new issue, please confirm:

- [X] I have [searched for duplicate or closed issues](https://github.com/aws-amplify/amplify-ui/issues?q=is%3Aissue+) and [discussions](https://github.com/aws-amplify/amplify-ui/discussions).
- [X] I have tried disabling all browser extensions or using a different browser
- [X] I have tried deleting the node_modules folder and reinstalling my dependencies
- [X] I have read the guide for [submitting bug reports](https://github.com/aws-amplify/amplify-ui/blob/main/CONTRIBUTING.md#bug-reports).

### On which framework/platform are you having an issue?

React Native

### Which UI component?

Authenticator

### How is your app built?

Create React App

### What browsers are you seeing the problem on?

Android (React Native)

### Please describe your bug.

From SignIn component:
- if I submit without username and password I get the error:  `Cannot read property 'username' of undefined`
- if I submit with username and without password I get the error:  `Custom auth lambda trigger is not configured for the user pool`

So I tried configuring both fields required, first with required prop, then with isRequird prop (please see following snipped) but it seems ignored.

I also tried overriding handleSignIn and returning false when the username or password are empty.
This indeed avoids submission but has 2 drawbacks:
- no message is displayed regarding the error (minor in this case)
- signin form becomes "grayed out", user cannot write nothing inside

### What's the expected behaviour?

When setting "isRequired" field property to true and the field is empty, the form should not be submitted

### Help us reproduce the bug!

See snippet below

### Code Snippet

```typescript
// Put your code below this line.
      <Authenticator
        components={{
          SignIn: ({fields, ...props}) => (
            <Authenticator.SignIn
              {...props}
              fields={fields.map(field => ({
                ...field,
                isRequired: true,
              }))}
            />
          ),
        }}>
        <SignOutButton />
      </Authenticator>
```

**Comment by ioanabrooks** [comment]
Addressed Cognito internal error messages being exposed, will follow up with improvements to required fields handling across web and mobile as separate ticket.

**Comment by andreav** [comment]
Hello, I tried the fix in version 1.2.5

I can confirm the fix works when username is empty.

However there is still the problem when username is valorized but password is empty, with the same error: `Custom auth lambda trigger is not configured for the user pool`.

Moreover I noticed a problem also on new account registration.
If I switch to "Create Account" and just press "Create Acocunt" button without compiling any field, i get the error: `right operand of 'in' is not an object`
I can open a separate issue if you prefer.
